src/classLibrary/ParsingStrategies/GuidResolver.ts

```diff
@@ -0,0 +1,96 @@
+import {
+    CyanSafe,
+    FileContent,
+    FileSystemInstance,
+    IGuidGenerator,
+    IParsingStrategy,
+    VirtualFileSystemInstance
+} from "../interfaces/interfaces";
+
+class GuidResolver implements IParsingStrategy {
+    private guidGenerator: IGuidGenerator;
+
+    constructor(guidGenerator: IGuidGenerator) {
+        this.guidGenerator = guidGenerator;
+    }
+
+    Count(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): Map<string, number> {
+        const guids: string[] = cyan.guid;
+        const result: Map<string, number> = new Map<string, number>();
+        virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            guids.Each((guid: string) => {
+                const count = this.CountKeyInVFS(guid, virtualFile);
+
+                if (result.has(guid)) {
+                    result.set(guid, result.get(guid)! + count);
+                } else {
+                    result.set(guid, count);
+                }
+            });
+        });
+
+        return result;
+    }
+
+    private CountKeyInVFS(key: string, virtualFile: VirtualFileSystemInstance): number {
+        return VirtualFileSystemInstance.match(virtualFile, {
+            File: (file: FileSystemInstance) => {
+                let tempCount = 0;
+                if (file["content"] != null && file.ignore.guidResolver.content) {
+                    FileContent.if.String(file.content, (str) => {
+                        tempCount += str.toUpperCase().Count(key.toUpperCase());
+                    });
+                }
+                return tempCount;
+            },
+            default: () => 0,
+        });
+    }
+
+    ResolveFiles(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+        return virtualFiles;
+    }
+
+    ResolveContents(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+        const guidsMap: Map<string, string> = cyan.guid.AsKey(() => this.guidGenerator.GenerateGuid());
+        return virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            VirtualFileSystemInstance.match(virtualFile, {
+                File: (file: FileSystemInstance) => {
+                    if (!file.ignore.guidResolver.content) return;
+                    if (file["content"] == null) return;
+                    FileContent.if.String(file.content, (str: string) => {
+                        guidsMap
+                            .Each((k: string, v: string) => {
+                                str = str.ReplaceAll(k.toLowerCase(), v).ReplaceAll(k.toUpperCase(), v);
```

Comment:
What's the reason to `replaceAll` twice?

Reply:
Is it to ensure that all sample GUIDs that are either lower or upper case are replaced accordingly?

Reply:
yeah right now the matching is case insensitive so regardless of the GUID's case, it will be replaced